core/src/bms/player/beatoraja/select/BarRenderer.java

```diff
@@ -407,6 +407,93 @@ public Map<String, Object> getFilter() {
 		public void setFilter(Map<String, Object> filter) {
 			this.filter = filter;
 		}
+
+		public Boolean filterSong(ScoreData scoreData) {
+			Set<String> filterKey = this.getFilter().keySet();
+			for (String key : filterKey) {
+				String getterMethodName = "get" + key.substring(0, 1).toUpperCase()
+						+ key.substring(1);
+				try {
+					if (this.getFilter().get(key) instanceof Integer) {
+						Integer value = (Integer)this.getFilter().get(key);
+						if (scoreData == null) {
+							if (0 != value) {
+								return false;
+							}
+						} else {
+							Method getterMethod = ScoreData.class.getMethod(getterMethodName);
+							Object propertyValue = getterMethod.invoke(scoreData);
+							if (!propertyValue.equals(value)) {
+								return false;
+							}
+						}
+						return true;
+					}
+					Object valueArr[] = ((String)this.getFilter().get(key)).split("&&");
+					for (Object value : valueArr)
+					{
+						value = ((String)value).replaceAll("\\s","");
+						if (scoreData == null) {
+							if (value instanceof String) {
```

Comment:
As you cast value to a String on line 435 this if statement will always return True, were you intending to catch Null's here? Not sure I understand

Reply:
That is a left-over from other approach, thanks for noting it.